test/cartesian/ErrorBar.spec.tsx

```diff
@@ -215,4 +217,170 @@ describe('<ErrorBar />', () => {
       transform: 'scaleY(1)',
     });
   });
+
+  test.each(['category', undefined])('throws when direction=x and XAxis id type=%s', (domainType: AxisDomainType) => {
+    expect(() => {
+      render(
+        <BarChart data={barData} width={500} height={500}>
+          <XAxis dataKey="name" type={domainType} />
+          <Bar isAnimationActive={false} dataKey="uv">
+            <ErrorBar isAnimationActive animationEasing="linear" dataKey="uvError" direction="x" />
+          </Bar>
+        </BarChart>,
+      );
+    }).toThrow('Invariant failed: ErrorBar requires Axis type property to be "number".');
+  });
+
+  describe('ErrorBar and axis domain interaction', () => {
+    test('ErrorBar should extend YAxis domain', () => {
+      const { container, rerender } = render(
+        <BarChart data={barData} width={500} height={500}>
+          <YAxis dataKey="uv" />
+          <Bar isAnimationActive={false} dataKey="uv" />
+        </BarChart>,
+      );
+
+      assertErrorBars(container, 0);
+      expectYAxisTicks(container, [
+        {
+          textContent: '0',
+          x: '57',
+          y: '495',
+        },
+        {
+          textContent: '850',
+          x: '57',
+          y: '372.5',
+        },
+        {
+          textContent: '1700',
+          x: '57',
+          y: '250',
+        },
+        {
+          textContent: '2550',
+          x: '57',
+          y: '127.5',
+        },
+        {
+          textContent: '3400',
+          x: '57',
+          y: '5',
+        },
+      ]);
+      rerender(
+        <BarChart data={barData} width={500} height={500}>
+          <YAxis dataKey="uv" />
+          <Bar isAnimationActive={false} dataKey="uv">
+            <ErrorBar isAnimationActive animationEasing="linear" dataKey="uvError" />
+          </Bar>
+        </BarChart>,
+      );
+      assertErrorBars(container, 4);
+      // these ticks are in the same position but going higher because the ErrorBar pushed YAxis domain
+      expectYAxisTicks(container, [
+        {
+          textContent: '0',
+          x: '57',
+          y: '495',
+        },
+        {
+          textContent: '900',
+          x: '57',
+          y: '372.5',
+        },
+        {
+          textContent: '1800',
+          x: '57',
+          y: '250',
+        },
+        {
+          textContent: '2700',
+          x: '57',
+          y: '127.5',
+        },
+        {
+          textContent: '3600',
+          x: '57',
+          y: '5',
+        },
+      ]);
+    });
+
+    test('ErrorBar not extend XAxis domain - this looks like a bug to me!', () => {
```

Comment:
Interesting, I found https://github.com/recharts/recharts/issues/634 which has a screenshot of the exact same problem and says it's fixed. I will backport the same test to 2.x branch to see where it has regressed.

Reply:
Hmmmmm good catch

Reply:
Okay the exact same problem is on master: 
<img width="487" alt="image" src="https://github.com/recharts/recharts/assets/1100170/acd4a98a-5913-4d35-9851-b05bd11bc182">

Reply:
Good to know